### ISSUE TITLE
EfficientDet-D0(with Weight)   Object detection results are too bad

### ISSUE THREAD

**Issue by opentld** [toandaominh1997/EfficientDet.Pytorch#90]
Download checkpoint_VOC_efficientdet-d0_268.pth from this link: https://drive.google.com/file/d/1r7MAyBfG5OK_9F_cU8yActUWxTHOuOpL/view?usp=sharing  
,The results of object detection is fearsome.
As shown below
![image](https://user-images.githubusercontent.com/21096515/72248962-59708600-3633-11ea-8259-f633e53c7ae4.png)

Do I have to retrain the model?

**Comment by opentld** [comment]
@toandaominh1997

**Comment by tienthegainz** [comment]
Try to increase the threshold

**Comment by immarioo** [comment]
i met same error, i set the threshold like 0.98, it still detects many boxes.
![图片](https://user-images.githubusercontent.com/12837585/72412935-9199d500-37a9-11ea-9771-2f5d66545477.png)
and i trained my own voc datasets for 70 epochs and the map is 0.  someone know how to fix it?

**Comment by CraigWang1** [comment]
The number next to the label name is the confidence score, and most of the boxes that you are displaying only have accuracy scores of 1-2%. To see the relevant predictions, just display the bboxes with >80%.

So, I guess you might need to change the source code in demo.py

Hope this helps.

**Comment by xafarranxera** [comment]
That's true. The problem is that no filtering is performed in demo.py. After appending boxes, labels and scores, add this:
```
dataframe = list(zip(bboxes, labels, bbox_scores))
dataframe = list(filter(lambda x: x[2] >= args.threshold*100, dataframe))

if(len(dataframe) > 0):
    bboxes, labels, bbox_scores = list(zip(*dataframe))
else:
    bboxes, labels, bbox_scores = [], [], []
```
Best wishes.

**Comment by opentld** [comment]
The problem has been fixed! just add codes like this below : 
![image](https://user-images.githubusercontent.com/21096515/73149213-25996400-40fb-11ea-8904-7bfbbf18a7cc.png)

and the result like this:
![image](https://user-images.githubusercontent.com/21096515/73149269-7446fe00-40fb-11ea-8910-384acef8867f.png)

although, the BGR or RBG channels might be some problem remain, but the process is right !

Thanks @xafarranxera  a lot !

**Comment by xafarranxera** [comment]
Hello again, you're welcome. Regarding BGR problem, you just have to add a single code line at the beginning, as stated in #96 issue.

**Comment by minji-o-j** [comment]
Hello do you still have the .pth file ?
I need to download that file, but the link doesn't work..
